homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -153,6 +153,15 @@ def convert_and_round(
         state_class=SensorStateClass.MEASUREMENT,
         is_available=lambda v: v.is_lsc_enabled and v.has_combustion_drivetrain,
     ),
+    BMWSensorEntityDescription(
+        key="activity",
+        translation_key="climate_status",
+        key_class="climate",
+        device_class=SensorDeviceClass.ENUM,
+        options=CLIMATE_ACTIVITY_STATE,
```

Comment:
Maybe loop through `bimmer_connected.vehicle.climate.ClimateActivityState?  instead so we don't have to specify a separate enum?

`[s.lower() for s in ClimateActivityState if s != ClimateActivityState.UNKNOWN]`

Or is this something we shouldn't do in HA?

Reply:
We shouldn't do this in HA. If that enum changes in a dependency bump, the integration is missing a translation key, so we have this kind of guideline to have these lists in the integration itself

Reply:
i have no idea what is the right way, just pushed your suggestion, thanks

Reply:
ok, will revert then, sorry, thanks @joostlek

Reply:
@joostlek @rikroe reverted, rebased and pushed now. thanks

Reply:
Ah sorry I seem to have created more confusion.

I see the issue with a dependency bump, but what would happen if a new enum state is added without changing the code here?

Would there be a hard error vs a missing translation? Of so, I would rather have the missing translation than breaking the whole sensor.

Reply:
I am willing to argue with others that if you use a snapshot test to snapshot the whole state/entity registry entry (which includes options) we can just use the enum, because if you then test the content of the options field, but we can maybe do that in a folowup to get this sensor in first :)

Reply:
Sounds reasonable! Sensors are already in the snapshot test with options. 

@brave0d you'll have to update the pytest snapshot by running ` pytest --snapshot-update tests/components/bmw_connected_drive/test_sensor.py` and then checking in the changed `test_sensor.ambr`.

It only didn't run (and therefore didn't fail) because this seems to be your first PR to HA (and therefore the CI run needs to be approved every time).

Reply:
@rikroe yes, it's my first PR and I am not sure what i am doing wrong - trying to update snapshots and getting the following errors 🤦 :

```
==================================================================== short test summary info =====================================================================
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_entity_state_attrs - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_range_total-unit_system0-279-km] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_range_total-unit_system1-173.36-mi] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_mileage-unit_system2-137009-km] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_mileage-unit_system3-85133.45-mi] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_battery_percent-unit_system4-82-%] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_battery_percent-unit_system5-82-%] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_range_electric-unit_system6-174-km] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_range_electric-unit_system7-108.12-mi] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_fuel-unit_system8-6-L] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_fuel-unit_system9-1.59-gal] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_range_fuel-unit_system10-105-km] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.i3_rex_remaining_range_fuel-unit_system11-65.24-mi] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.m340i_xdrive_remaining_fuel_percent-unit_system12-80-%] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'
FAILED tests/components/bmw_connected_drive/test_sensor.py::test_unit_conversion[sensor.m340i_xdrive_remaining_fuel_percent-unit_system13-80-%] - AttributeError: module 'homeassistant.components' has no attribute 'zeroconf'

Results (0.75s):
      15 error
```

Reply:
In case you haven't set up the full dev environment, you would need to do that (see https://developers.home-assistant.io/docs/development_environment).

I personally don't use the Dev Containers, but just a Python 3.12 venv and then inside the venv run `script/setup´ and `pip install -r requirements_test.txt bimmer-connected`. Then pytest should run.

Reply:
@brave0d just created a PR against your branch (https://github.com/brave0d/core/pull/1) that has the snapshot changes in. Just merge it and then @joostlek can contine here.

Also had to add a second variable to the lambda function (which I missed before).